src/main/scala/li/cil/oc/server/machine/Machine.scala

```diff
@@ -1070,6 +1172,21 @@ object Machine extends MachineAPI {
     val Running = Value("Running")
   }
 
+  /** Possible sync-related states of the computer */
+  private object SyncState extends Enumeration {
+    /** The computer is in the Insecure state. It is asynchronous and, unless "*" is whitelisted, cannot leave this state without a reboot. */
+    val Insecure = Value("Insecure")
+
+    /** The computer is in the Insecure state, and is synchronous. This state is only permitted if "*" is whitelisted, and will decay to Insecure if "*" is removed from the whitelist. */
+    val InsecureSync = Value("InsecureSync")
+
+    /** The computer is in the Secure state, and is currently asynchronous. */
+    val SecureAsync = Value("SecureAsync")
+
+    /** The computer is in the Secure state, and is currently synchronous. */
+    val SecureSync = Value("SecureSync")
+  }
+
   /** Signals are messages sent to the Lua state from Java asynchronously. */
```

Comment:
I have some opinions about the naming of values here and the joining of two orthagonal ideas. I would prefer two separate enums, Insecure/Secure and Async/Sync
Also, I would prefer Untrusted/Trusted in place of Insecure/Secure
I would like you to genuinely consider this separation of enums as it would simplfy the code. Give some consideration for the name change as I feel it makes it more clear.